main.py

```diff
@@ -13,74 +16,65 @@
 
 
 def main():
+    parser = argparse.ArgumentParser(description="Check how much time I waste on this playlist")
+
+    parser.add_argument("-p", "--playlist", help="YouTube video playlist url", action=YoutubeURLAction, required=True)
+    parser.add_argument("-r", "--range",
+                        help="Video range to select (example: -r 1,3-5,7+9+10). If not set, all videos are selected",
+                        action=RangeAction)
+
+    args = parser.parse_args()
+
+    logger = Logger()
+
     try:
-        logger = Logger()
         logger.success("How much time I waste on this playlist was started")
 
-        playlist_url = extract_url(sys.argv)
-        playlist_id = extract_playlist_id(playlist_url)
-
-        video_ids = retrieve_video_ids(playlist_id)
+        video_ids = retrieve_video_ids(args.playlist, args.range)
```

Comment:
Mh finde es nicht so cool wenn eine Funktion alles macht.
Clean Code passt da dann auch nicht mehr wirklich.

Eine Funktion sollte soweit es geht nur 1 Aufgabe übernehmen, aber hier packst du soweit ich das sehe vieles in eine Funktion rein.
Selbst mein Code war noch optimierungsbedürftig was clean code angeht aber ja.